pkg/ratelimit/controller.go

```diff
@@ -27,12 +30,51 @@ type Controller struct {
 	limiters sync.Map
 	// the label which is in labelAllowList won't be limited, and only inited by hard code.
 	labelAllowList map[string]struct{}
+
+	ctx              context.Context
+	cancel           context.CancelFunc
+	apiType          string
+	concurrencyGauge *prometheus.GaugeVec
 }
 
 // NewController returns a global limiter which can be updated in the later.
-func NewController() *Controller {
-	return &Controller{
-		labelAllowList: make(map[string]struct{}),
+func NewController(ctx context.Context, typ string, concurrencyGauge *prometheus.GaugeVec) *Controller {
+	ctx, cancel := context.WithCancel(ctx)
+	l := &Controller{
+		ctx:              ctx,
+		cancel:           cancel,
+		labelAllowList:   make(map[string]struct{}),
+		apiType:          typ,
+		concurrencyGauge: concurrencyGauge,
+	}
+	if concurrencyGauge != nil {
+		go l.collectMetrics()
+	}
+	return l
+}
+
+func (l *Controller) Close() {
+	l.cancel()
+}
+
+func (l *Controller) collectMetrics() {
+	tricker := time.NewTicker(time.Second * 5)
+	defer tricker.Stop()
+	for {
+		select {
+		case <-l.ctx.Done():
+			return
+		case <-tricker.C:
+			l.limiters.Range(func(key, value any) bool {
+				limiter := value.(*limiter)
+				label := key.(string)
+				// Due to not in hot path, no need to save sub Gauge.
+				if con := limiter.getConcurrencyLimiter(); con != nil {
+					l.concurrencyGauge.WithLabelValues(l.apiType, label).Set(float64(con.getCurrent()))
```

Comment:
Maybe we can find another way to show the statistics, the current implementation only captures the instant value every 15s.